src/smc.jl

```diff
@@ -128,3 +137,48 @@ function AbstractMCMC.step(
 
     return PGSample(newtrajectory, logevidence), PGState(newtrajectory)
 end
+
+function AbstractMCMC.step(
+    rng::Random.AbstractRNG, model::AbstractMCMC.AbstractModel, sampler::PGAS; kwargs...
+)
+    # Create a new set of particles.
+    particles = ParticleContainer(
+        [SSMTrace(typeof(model)(), TracedRNG()) for _ in 1:(sampler.nparticles)],
+        TracedRNG(),
+        rng,
+    )
+
+    # Perform a particle sweep.
+    logevidence = sweep!(rng, particles, sampler.resampler)
+
+    # Pick a particle to be retained.
+    trajectory = rand(particles.rng, particles)
+    return PGSample(trajectory, logevidence), PGState(trajectory)
+end
+
+function AbstractMCMC.step(
+    rng::Random.AbstractRNG,
+    model::AbstractStateSpaceModel,
+    sampler::PGAS,
+    state::PGState;
```

Comment:
Maybe consider using the argument signature below, so that we can merge these two `AbstractMCMC.step` implementations? 

```suggestion
    state::Union{PGState,nothing} = nothing;
```

Reply:
I also changed the signature of `PG`, most of the code was duplicated